ui/apps/platform/src/Containers/Clusters/DelegateScanning/DelegatedImageScanningForm.tsx

```diff
@@ -0,0 +1,160 @@
+
+export type DelegatedImageScanningFormProps = {
+    delegatedRegistryClusters: DelegatedRegistryCluster[];
+    delegatedRegistryConfig: DelegatedRegistryConfig;
+    isEditing: boolean;
+    setDelegatedRegistryConfig: (delegatedRegistryConfig: DelegatedRegistryConfig) => void;
+    setIsNotEditing: () => void;
+};
+
+function DelegatedImageScanningForm({
+    delegatedRegistryClusters,
+    delegatedRegistryConfig,
+    isEditing,
+    setDelegatedRegistryConfig,
+    setIsNotEditing,
+}) {
+    const [errorMessage, setErrorMessage] = useState('');
+
+    const {
+        dirty,
+        // errors,
+        isSubmitting,
+        isValid,
+        resetForm,
+        setFieldValue,
+        setSubmitting,
+        submitForm,
+        values,
+    } = useFormik<DelegatedRegistryConfig>({
+        initialValues: delegatedRegistryConfig,
+        // validationSchema,
+        onSubmit: (valuesToSubmit) => {
+            updateDelegatedRegistryConfig(valuesToSubmit)
+                .then((delegatedRegistryConfigUpdated) => {
+                    // Reset form state from response although in theory, same as payload.
+                    resetForm({ values: delegatedRegistryConfigUpdated });
+                    // Because Page renders Form after initial request,
+                    // the following update to its state, although consistent,
+                    // seems redundant with update of form state.
+                    setDelegatedRegistryConfig(delegatedRegistryConfigUpdated);
+                    setIsNotEditing();
+                })
+                .catch((error) => {
+                    setErrorMessage(getAxiosErrorMessage(error));
+                })
+                .finally(() => {
+                    setSubmitting(false);
+                });
+        },
+    });
+
+    function onCancel() {
+        resetForm();
+        setIsNotEditing();
+    }
+
+    function addRegistry() {
+        return setFieldValue('registries', [...values.registries, { path: '', clusterId: '' }]);
+    }
+
+    function deleteRegistry(indexToDelete) {
+        return setFieldValue(
+            'registries',
+            values.registries.filter((_, index) => index !== indexToDelete)
+        );
+    }
+
+    function setRegistryClusterId(indexToSet: number, clusterId: string) {
+        return setFieldValue(
+            'registries',
+            values.registries.map((registry, index) =>
+                index === indexToSet ? { path: registry.path, clusterId } : registry
+            )
+        );
+    }
+
+    function setRegistryPath(indexToSet: number, path: string) {
+        return setFieldValue(
+            'registries',
+            values.registries.map((registry, index) =>
+                index === indexToSet ? { path, clusterId: registry.clusterId } : registry
+            )
+        );
+    }
+
+    return (
+        <Flex direction={{ default: 'column' }}>
+            {errorMessage.length !== 0 && (
```

Comment:
Nit: What do you think about the errorMessage state being `string | null` instead of only `string`? It seems very unlikely, but if the server returned an error with an empty string as the message it could obscure that there was a failure in the UI.

Reply:
Yes, indeed. I will update after initial CI passes.